move get_configuration_parameters from submission to configuration

superstar54 opened the pull request:
- move get_configuration_parameters from submission to configuration
- link the value of the `configuration  step` to the `input_parameters` of the submit step

unkcpz commented on tests/conftest.py:
```diff
@@ -209,57 +219,51 @@ def submit_step_widget_generator(
+        configure_step.advanced_settings.smearing.degauss.value = degauss
+        configure_step.confirm()
+        #
+        submit_step = app.steps[2][1]
```
```suggestion
        submit_step = app.submit_qe_app_work_chain_step
```

unkcpz commented on src/aiidalab_qe/app/submission/__init__.py:
```diff
@@ -399,136 +378,38 @@ def submit(self, _=None):
-            electronic_type=ElectronicType(parameters.electronic_type),
-            overrides=parameters.overrides,
-            initial_magnetic_moments=parameters.initial_magnetic_moments,
+            protocol=parameters["protocol"],
```
As we discussed offline, we agree keep on using dataclass is better for the moment.

superstar54 replied:
After another discussion, it is better to use `dict` directly. Because the `value` of the configure step is now a traitlet and linked to the `input_parameters` (also a trailet) of the submit step. 

```
    # output dictionary
    value = tl.Dict()
```

unkcpz replied:
Can we name it `configuration_parameters` as the function name has. The `value` of the step makes no much sense and is confusing.

superstar54 replied:
yes, i changed it to `configuration_parameters`

unkcpz reviewed (changes requested):
Thanks! Two minor comments.

superstar54 commented:
@unkcpz Thanks for the comments. I updated:
- In the report, remove `if` and use `.get(key, None)`, and remove all `None` values in the final step.
- Use the name of the step directly instead of steps[0][1]. Also, I renamed the step using a short name.

Please have another look.

unkcpz commented on src/aiidalab_qe/app/main.py:
```diff
@@ -143,16 +133,16 @@ def _observe_process_selection(self, change):
+                    self.configure_step.state = WizardAppWidgetStep.State.SUCCESS
+                    self.submit_step.process = process
+
+    @property
+    def steps(self):
+        return self._wizard_app_widget.steps
```
Not sure I understand this. Where is this property used?

superstar54 replied:
It was used for tests before, e.g. self.steps[1][1]. Now, we can use self.configure_step directly. So I removed it.

unkcpz commented on src/aiidalab_qe/app/result/report.py:
```diff
@@ -27,47 +27,49 @@ def _generate_report_dict(builder_parameters: dict):
-    yield "relaxed", builder_parameters["run_relax"]
-    yield "bands_computed", builder_parameters["run_bands"]
-    yield "pdos_computed", builder_parameters["run_pdos"]
+    yield "relax_method", builder_parameters.get("relax_type", None)
+    yield "relaxed", builder_parameters.get("run_relax", None)
+    yield "bands_computed", builder_parameters.get("run_bands", None)
+    yield "pdos_computed", builder_parameters.get("run_pdos", None)
```
The `None` is the default return value if the key not defined when using `.get`. So just do `builder_parameters.get("relax_type")`

superstar54 replied:
:+1: , i removed the `None`.

unkcpz commented on src/aiidalab_qe/app/result/report.py:
```diff
@@ -91,6 +93,7 @@ def _fmt_yes_no(truthy):
     )
     template = resources.read_text(static, "workflow_summary.jinja")
     style = resources.read_text(static, "style.css")
+    report = {key: value for key, value in report.items() if value is not None}
```
:+1:

unkcpz commented on tests/test_submit_qe_workchain.py:
```diff
@@ -47,11 +46,11 @@ def test_create_builder_default(
-    if "initial_magnetic_moments" not in builder_parameters:
-        parsed = BeautifulSoup(report_html, "html.parser")
-        assert parsed.find("initial_magnetic_moments").text == "None"
+    # All None value is removed in the final report
+    assert report_html.count("None") == 0
+    # if "initial_magnetic_moments" not in builder_parameters:
+    #     parsed = BeautifulSoup(report_html, "html.parser")
+    #     assert parsed.find("initial_magnetic_moments").text == "None"
```
This actually change the behaviour of the original test and skipped the thing we want to test initially, that when the initial_magnetic_moment not set the report dict will not contain it.

Instead, of the changes, you can call `_generate_report_dict` and still have this behaviour tested.

superstar54 replied:
I find that there is nothing needs to be changed here, so I revert it.

unkcpz reviewed (changes requested):
@superstar54 Thanks! Just minor requests.

superstar54 commented:
> @superstar54 Thanks! Just minor requests.

Hi @unkcpz  Thanks! I changed as you suggested.

unkcpz reviewed (approved):
All good! thanks!
